File: 02_Angular/api-pokemon/src/app/services/pokemon-service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { PokemonListResponse } from '../interfaces/pokemon-list-response.interface';

@Injectable({
  providedIn: 'root',
})
export class PokemonService {
  constructor(private http: HttpClient) {}

  getPokemonList(): Observable<PokemonListResponse> {
    return this.http.get<PokemonListResponse>('https://pokeapi.co/api/v2/pokemon');
  }
}

File: 04_Flutter/flutter_api/lib/core/services/pokemon_service.dart
import 'package:flutter_api/core/interfaces/pokemon_interface.dart';
import 'package:flutter_api/core/models/pokemon_list_response.dart';
import 'package:http/http.dart' as http;

class PokemonService implements PokemonInterface {
  final String _apiBaseUrl = "https://pokeapi.co/api/v2/pokemon/";

  @override
  Future<List<PokemonResult>> getAllPokemon() async {
    var url = Uri.parse(_apiBaseUrl + '?limit=200&offset=0');
    try {
      var response = await http.get(url);

      if (response.statusCode == 200 || response.statusCode == 201) {
        /*var pokemonListResponse = PokemonListResponse.fromJson(
          jsonDecode(response.body),
        );*/
        var pokemonListResponse = PokemonListResponse.fromJson(response.body);
        return pokemonListResponse.results;
      } else {
        return [];
      }
    } catch (e) {
      return [];
    }
  }

  @override
  Future<PokemonResult> getOnePokemon(String id) {
    // TODO: implement getOnePokemon
    throw UnimplementedError();
  }
}
